### PR TITLE
MAINT: Apply assorted ruff/flake8-pie rules (PIE)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
This will help add more rule after #27300.

### CODE REVIEW

File: numpy/_core/tests/test_numeric.py [hunk 1]
```diff
@@ -4141,14 +4141,14 @@ def test_broadcast_error_kwargs(self):
         #gh-13455
         arrs = [np.empty((5, 6, 7))]
         mit  = np.broadcast(*arrs)
-        mit2 = np.broadcast(*arrs, **{})
+        mit2 = np.broadcast(*arrs)
```
Comment (seberg) [inline numpy/_core/tests/test_numeric.py:4144]: This test is maybe not particularly important but covers a C code branch and is certainly intentional.

**Review by seberg — APPROVED** [review]
LGTM, although I would slightly prefer to just ignore that one fix, since it is intentional.

File: numpy/_core/tests/test_numeric.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply numpy/_core/tests/test_numeric.py]: Fixed.

File: numpy/_build_utils/tempita/_tempita.py [hunk 2]
```diff
@@ -784,7 +784,7 @@ def parse_expr(tokens, name, context=()):
     expr = expr.strip()
     if expr.startswith("py:"):
         expr = expr[3:].lstrip(" \t")
-        if expr.startswith("\n") or expr.startswith("\r"):
+        if expr.startswith(("\n", "\r")):
```
Comment (seberg) [inline numpy/_build_utils/tempita/_tempita.py:787]: I will just ignore it, but this is vendor'd code and style fixes seem odd.  When you run these tools, do you use `.gitattributes` to ignore vendor'd code?  (and should we add tempita there if it is helpful for such tools at least.)

Comment (DimitriPapadopoulos) [reply numpy/_build_utils/tempita/_tempita.py:787]: As far as I know, ruff does not read `.gitattributes`. Is marking vendored files in `.gitattributes` standard practice? Which tools use it?

I can revert changes on vendored code. On the other hand, tempita is unmaintained as far as I know, it doesn't make much sense to struggle to keep consistence with upstream.

Comment (seberg) [reply numpy/_build_utils/tempita/_tempita.py:787]: yeah, dunno if it is worth it.  Github uses the info, I am not sure what other tools do.

**Pull request merged**